### PR TITLE
Simplify new line indicator and message badge

### PR DESCRIPTION
**Pull request by marcaaron**
### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/224876

### Tests
1. Start a chat between User A and User B.
1. Ensure User A is not looking at the chat.
1. Send a message as User B
1. Verify the LHN row shows in bold for User A to indicate unread messages
1. Visit the chat as User A and verify the LHN row no longer shows in bold
1. Verify there is a “----- New” line above the message sent by User B
1. Scroll up and verify that the “New Messages” button appears floating above the content.
1. Verify that tapping this button will scroll you to the bottom of the chat but not clear the “---- New” line marker.
1. Mark a comment as “unread”
1. Verify the “---- New” line marker appears
1. Mark as unread comments above and below the existing new marker
1. Verify the marker appears in the correct position each time
1. While the new marker is active leave a comment as User A
1. Verify the new marker and new messages badge both disappear
1. Mark another comment as unread.
1. Navigate away from the chat
1. Navigate back to the chat
1. Verify the chat in the LHN shows as read
1. Verify the New Line marker and new messages badge is still present
1. Navigate away one more time and back
1. Verify the new marker and new messages badge both disappear
1. Mark another comment as unread.
1. Navigate away from the chat
1. Navigate back to the chat
1. Minimize the app and reopen it
1. Verify the new marker and new message badge both disappear
1. Switch to a mobile view (small screen) and mark a message as unread
1. Tap the header to reveal the LHN
1. Verify the chat in the LHN shows as “unread”
1. Tap the chat
1. Verify the new marker and badge are present
1. Tap the header to reveal the LHN
1. Verify the chat in the LHN shows as “read”
1. Tap the chat
1. Verify the new marker and badge are no longer present.
1. Scroll up in the chat so that last comment is out of view
1. Leave a comment as User B
1. Verify the “new messages badge” shows for User A
1. Scroll down and verify the “new line marker” shows for User A in the correct position.
1. Scroll back up and leave another comment as User B
1. Scroll down as User A
1. Verify the new marker is in the same position as it was before

- [ ] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

Same as Test Steps

- [ ] Verify that no errors appear in the JS console

### CODE REVIEW

**Comment by marcaaron** [comment]
[This PR](https://github.com/Expensify/Web-Expensify/pull/34696) is required to test atm as I discovered a regression caused by the `OpenReport` changes.

**Comment by marcaaron** [comment]
This is still on HOLD but testing well so marking it as ready.

File: src/pages/home/report/FloatingMessageCounter/index.js [hunk 1]
```diff
@@ -11,14 +11,8 @@ import withLocalize, {withLocalizePropTypes} from '../../../../components/withLo
 import FloatingMessageCounterContainer from './FloatingMessageCounterContainer';
 
 const propTypes = {
-    /** Count of new messages to show in the badge */
-    count: PropTypes.number,
-
     /** Whether the marker is active */
```
Comment (cead22) [inline src/pages/home/report/FloatingMessageCounter/index.js:14]:
```suggestion
    /** Whether the New Messages indicator is active */
```
Should we rename this file since it's no longer showing counts?

File: src/pages/home/report/FloatingMessageCounter/index.js [hunk 2]
```diff
@@ -11,14 +11,8 @@ import withLocalize, {withLocalizePropTypes} from '../../../../components/withLo
...
-    active: PropTypes.bool,
-
-    /** Callback to be called when user closes the badge */
-    onClose: PropTypes.func,
+    isActive: PropTypes.bool,
 
     /** Callback to be called when user clicks the marker */
```
Comment (cead22) [inline src/pages/home/report/FloatingMessageCounter/index.js:17]:
```suggestion
    /** Callback to be called when user clicks the New Messages indicator */
```
Not sure if these comments were copy paste from the new messages marker, or if we're calling both things a marker, but I think it's good to differentiate for clarity

File: src/pages/home/report/FloatingMessageCounter/index.js [hunk 3]
```diff
@@ -93,24 +85,10 @@ class FloatingMessageCounter extends PureComponent {
...
-                                            'newMessageCount',
-                                            {count: this.props.count},
-                                        )}
+                                        {this.props.translate('newMessages')}
                                     </Text>
                                 </View>
                             )}
-                            shouldRemoveRightBorderRadius
-                        />
-                        <Button
```
Comment (cead22) [inline src/pages/home/report/FloatingMessageCounter/index.js:105]: Does this mean the new messages indicator can't be closed with an `X`? Why the change? Just curious

File: src/pages/home/report/ReportActionsList.js [hunk 4]
```diff
@@ -134,8 +134,8 @@ class ReportActionsList extends React.Component {
         item,
         index,
     }) {
-        const shouldDisplayNewIndicator = this.props.report.newMarkerSequenceNumber > 0
-            && item.action.sequenceNumber === this.props.report.newMarkerSequenceNumber;
+        const shouldDisplayNewIndicator = this.props.newMarkerSequenceNumber > 0
```
Comment (cead22) [inline src/pages/home/report/ReportActionsList.js:137]: Is the first condition needed here? can both `sequenceNumber` and `newMarkerSequenceNumber` be equal to zero simultaneously?

Comment (cead22) [reply src/pages/home/report/ReportActionsList.js:137]: It looks like we explicitly set this to zero, could we set to null to indicate the lack of a value here instead? NAB but since zero could be a valid sequence number (just based on the data type), null could be a little cleaner

File: src/pages/home/report/ReportActionsView.js [hunk 5]
```diff
@@ -174,53 +176,79 @@ class ReportActionsView extends React.Component {
     }
 
     componentDidUpdate(prevProps) {
+        const isReportFullyVisible = this.getIsReportFullyVisible();
+
+        // NETWORK CHANGE DETECTED
```
Comment (cead22) [inline src/pages/home/report/ReportActionsView.js:181]: Not a fan of these capitalized comments, can we make them lowercase and keep consistency with all other comments, or use variable to communicate this? Eg,
```js
const bool wasNetworkChangeDetected = lodashGet(prevProps.network, 'isOffline') && !lodashGet(this.props.network, 'isOffline');
...
if (wasNetworkChangeDetected) {
...
```

File: src/pages/home/report/ReportActionsView.js [hunk 6]
```diff
@@ -174,53 +176,79 @@ class ReportActionsView extends React.Component {
...
+        // NEW REPORT ACTION APPEARED
         if (previousLastSequenceNumber !== currentLastSequenceNumber) {
             const lastAction = CollectionUtils.lastItem(this.props.reportActions);
             const isLastActionFromCurrentUser = lodashGet(lastAction, 'actorEmail', '') === lodashGet(this.props.session, 'email', '');
+
+            // NEW REPORT ACTION APPEARED FROM CURRENT USER
             if (isLastActionFromCurrentUser) {
-                // If a new comment is added and it's from the current user scroll to the bottom otherwise leave the user positioned where they are now in the list.
+                // If a new comment is added and it's from the current user scroll to the bottom otherwise leave the user positioned where
+                // they are now in the list.
```
Comment (cead22) [inline src/pages/home/report/ReportActionsView.js:205]: I think this comment should be above `if (isLastActionFromCurrentUser) {` and replace the all caps one

File: src/pages/home/report/ReportActionsView.js [hunk 7]
```diff
@@ -174,53 +176,79 @@ class ReportActionsView extends React.Component {
...
-            }
-
-            // When the last action changes, record the max action
-            // This will make the NEW marker line go away if you receive comments in the same chat you're looking at
-            if (isReportFullyVisible) {
-                Report.readNewestAction(this.props.report.reportID);
+                // If the current user sends a new message in the chat we clear the new marker since they have "read" the report
+                this.setState({newMarkerSequenceNumber: 0});
+            } else if (isReportFullyVisible) {
+                // NEW REPORT ACTION APPEARED FROM ANOTHER USER AND REPORT IS IN VIEW
```
Comment (cead22) [inline src/pages/home/report/ReportActionsView.js:211]:
```suggestion
```

File: src/pages/home/report/ReportActionsView.js [hunk 8]
```diff
@@ -174,53 +176,79 @@ class ReportActionsView extends React.Component {
...
+                    this.setState({newMarkerSequenceNumber: currentLastSequenceNumber});
+                }
             }
         }
 
-        // Update the new marker position and last read action when we are closing the sidebar or moving from a small to large screen size
-        if (isReportFullyVisible && reportBecomeVisible) {
-            this.updateNewMarkerPosition(this.props.report.unreadActionCount);
+        // REPORT PREVIOUSLY HIDDEN BY SIDEBAR/LHN OR VIEW EXPANDED FROM MOBILE TO DESKTOP LAYOUT
+        // We update the new marker position, mark the report as read, and fetch new report actions
```
Comment (cead22) [inline src/pages/home/report/ReportActionsView.js:225]:
```suggestion
        // If the report was previously hidden by the side bar, or the view is expanded from mobile to desktop layout
        // we update the new marker position, mark the report as read, and fetch new report actions
```

File: src/pages/home/report/ReportActionsView.js [hunk 9]
```diff
@@ -174,53 +176,79 @@ class ReportActionsView extends React.Component {
...
+            this.setState({
+                newMarkerSequenceNumber: this.props.report.unreadActionCount === 0
+                    ? 0
+                    : this.props.report.lastReadSequenceNumber + 1,
+            });
             Report.openReport(this.props.report.reportID);
         }
+
+        // REPORT PREVIOUSLY OPEN BUT USER NAVIGATED TO SIDEBAR/LHN
+        // When the user swipes back on the report or navigates to the LHN the ReportActionsView doesn’t unmount and just remains hidden.
```
Comment (cead22) [inline src/pages/home/report/ReportActionsView.js:238]: How's "swipes back on the report" different from "navigates to the LHN"?
```suggestion
        // When the user swipes back on the report or navigates to the LHN the ReportActionsView doesn't unmount and just remains hidden.
```

File: src/pages/home/report/ReportActionsView.js [hunk 10]
```diff
@@ -174,53 +176,79 @@ class ReportActionsView extends React.Component {
...
+        if (isReportFullyVisible && reportBecameVisible) {
+            this.setState({
+                newMarkerSequenceNumber: this.props.report.unreadActionCount === 0
+                    ? 0
+                    : this.props.report.lastReadSequenceNumber + 1,
+            });
             Report.openReport(this.props.report.reportID);
         }
+
+        // REPORT PREVIOUSLY OPEN BUT USER NAVIGATED TO SIDEBAR/LHN
```
Comment (cead22) [inline src/pages/home/report/ReportActionsView.js:237]:
```suggestion
```

File: src/pages/home/report/ReportActionsView.js [hunk 11]
```diff
@@ -174,53 +176,79 @@ class ReportActionsView extends React.Component {
...
+        // When the user swipes back on the report or navigates to the LHN the ReportActionsView doesn’t unmount and just remains hidden.
+        // The next time we navigate to the same report (e.g. by swiping or tapping the LHN row) we want the new marker to clear.
+        const sidebarOpened = !prevProps.isDrawerOpen && this.props.isDrawerOpen;
+        if (sidebarOpened && this.props.report.unreadActionCount === 0) {
+            this.setState({newMarkerSequenceNumber: 0});
+        }
+
+        // REPORT COMMENT MANUALLY MARKED AS UNREAD
+        // Checks to see if a report comment has been manually "marked as read". All other times when the lastReadSequenceNumber
```
Comment (cead22) [inline src/pages/home/report/ReportActionsView.js:246]:
```suggestion
        // Checks to see if a report comment has been manually "marked as unread". All other times when the lastReadSequenceNumber
```

File: src/pages/home/report/ReportActionsView.js [hunk 12]
```diff
@@ -174,53 +176,79 @@ class ReportActionsView extends React.Component {
...
+
+        // REPORT PREVIOUSLY OPEN BUT USER NAVIGATED TO SIDEBAR/LHN
+        // When the user swipes back on the report or navigates to the LHN the ReportActionsView doesn’t unmount and just remains hidden.
+        // The next time we navigate to the same report (e.g. by swiping or tapping the LHN row) we want the new marker to clear.
+        const sidebarOpened = !prevProps.isDrawerOpen && this.props.isDrawerOpen;
+        if (sidebarOpened && this.props.report.unreadActionCount === 0) {
+            this.setState({newMarkerSequenceNumber: 0});
+        }
+
+        // REPORT COMMENT MANUALLY MARKED AS UNREAD
```
Comment (cead22) [inline src/pages/home/report/ReportActionsView.js:245]:
```suggestion
```

**Review by cead22 — CHANGES_REQUESTED** [review]

File: src/pages/home/report/FloatingMessageCounter/index.js (hunk 3 above)
Comment (marcaaron) [reply src/pages/home/report/FloatingMessageCounter/index.js:105]: Largely just a matter of simplification and nobody fought to keep it. In the new design, the visibility of the component is enabled based on these conditions alone:

- The new marker is set
- The user is scrolled up past a certain distance

Doc here -> https://docs.google.com/document/d/1gvPDhGTxdcu-0lQxvruQnQJfJBrAmSjHieKciucV8eo/edit#bookmark=id.v1smzjvcr8ra

File: src/pages/home/report/ReportActionsList.js (hunk 4 above)
Comment (marcaaron) [reply src/pages/home/report/ReportActionsList.js:137]: Not sure if I understand the question. But the newMarkerSequenceNumber can be equal to zero and the created action has a sequenceNumber of 0. So we do this to avoid showing the marker for the "created" action. We don't want to show the marker when the last read sequenceNumber is 0.

As for using `null` instead, I don't have strong feelings on it, but maybe we can add a comment here at least to explain what I just mentioned.

File: src/pages/home/report/ReportActionsView.js (hunk 5 above)
Comment (marcaaron) [reply src/pages/home/report/ReportActionsView.js:181]: With the comments I wanted to create a sense that these are discrete "events" or "situations" we might encounter. 

I had a draft where each of these was moved to a method like `checkIfNetworkChangedAndRefetch()`, but that felt like something Uncle Bob would do and maybe overkill. I eventually just decided to do something obvious and ugly. Will try the variable thing! Thanks!

File: src/pages/home/report/ReportActionsView.js (hunk 9 above)
Comment (marcaaron) [reply src/pages/home/report/ReportActionsView.js:238]: I'll be more general here. I meant swipe or tap the back button. Both would be ways of navigating to the LHN.

File: src/pages/home/report/FloatingMessageCounter/index.js (hunk 3 above)
Comment (marcaaron) [reply src/pages/home/report/FloatingMessageCounter/index.js:105]: Do you think it's something we'll miss? Should be relatively easy to add it back in if so. I kinda found it just made the code harder to understand and not really all that useful of a feature.

File: src/pages/home/report/ReportActionsList.js (hunk 4 above)
Comment (marcaaron) [reply src/pages/home/report/ReportActionsList.js:139]: Let me know if this comment helps. Or if you feel strongly that we should use `null`. I'm inclined to leave it since everything is working well for me.

File: src/pages/home/report/FloatingMessageCounter/index.js (hunk 3 above)
Comment (cead22) [reply src/pages/home/report/FloatingMessageCounter/index.js:105]: No I like this change actually, this was just my first time looking at this code, and this is a change I "picked up" that I wasn't expecting, and I wanted to make sure I understood it correctly, since I'm still kind of a noob when it comes to reviews in App

File: src/pages/home/report/ReportActionsView.js (hunk 5 above)
Comment (cead22) [reply src/pages/home/report/ReportActionsView.js]: Making these comments not be all uppercase already solves my concern. I think we should follow our guidelines, so others can as well. If we break consistency for our comments, contributors may start doing the same

Comment (cead22) [reply src/pages/home/report/ReportActionsView.js]: And to be clear, I did appreciate the comments that are in this code, as it would've been much harder to follow without them

**Review by cead22 — APPROVED** [review]

**Review by amyevans — APPROVED** [review]
I don't have a lot of context for these changes, but they test well for me locally, and the logic and comments make sense to me

**Comment by roryabraham** [comment]
Going to try reverting this to see if it's the cause for https://github.com/Expensify/App/issues/10751
